runtime/compiler/runtime/JITServerAOTCache.cpp

```diff
@@ -435,74 +367,47 @@ CachedAOTMethod::create(const AOTCacheClassChainRecord *definingClassChainRecord
                                     records, code, codeSize, data, dataSize);
    }
 
-CachedAOTMethod *
-CachedAOTMethod::read(FILE *f,
-                      const Vector<AOTCacheClassLoaderRecord *> &classLoaderRecords,
-                      const Vector<AOTCacheClassRecord *> &classRecords,
-                      const Vector<AOTCacheMethodRecord *> &methodRecords,
-                      const Vector<AOTCacheClassChainRecord *> &classChainRecords,
-                      const Vector<AOTCacheWellKnownClassesRecord *> &wellKnownClassesRecords,
-                      const Vector<AOTCacheAOTHeaderRecord *> &aotHeaderRecords)
-   {
-   SerializedAOTMethod header;
-   if (1 != fread(&header, sizeof(header), 1, f))
-      return NULL;
-
-   if ((!header.isValid()) ||
-       (header.definingClassChainId() >= classChainRecords.size()) ||
-       (header.aotHeaderId() >= aotHeaderRecords.size()) ||
-       !classChainRecords[header.definingClassChainId()])
-      return NULL;
-
-   auto record = new (AOTCacheRecord::allocate(size(header.numRecords(), header.codeSize(), header.dataSize())))
-                     CachedAOTMethod(classChainRecords[header.definingClassChainId()]);
-   memcpy(&record->_data, &header, sizeof(header));
-
-   if (1 != fread(record->data().offsets(), sizeof(SerializedSCCOffset) * header.numRecords() + header.codeSize() + header.dataSize(), 1, f))
-      goto error;
-
-   for (size_t i = 0; i < header.numRecords(); ++i)
+bool
+CachedAOTMethod::setSubrecordPointers(JITServerAOTCacheReadContext &context)
+   {
+   for (size_t i = 0; i < data().numRecords(); ++i)
```

Comment:
Would be helpful to add a comment that the defining class chain record pointer is already set before this method is called.